src/components/OfferCheckoutCard/OfferCheckoutCard.tsx

```diff
@@ -115,8 +119,83 @@ const OfferCheckoutCard = () => {
     );
   };
 
+  const generateDescriptionForCoupon = () => {
```

Comment:
same here - possibly to be simplified

Reply:
Get rid of global `description` value, create `description` locally in every case and return whole translation. Like I proposed above renderCouponNote. We can also standarize those func names to `getXXXDescription` or `getXXXCopy`

Local description variable example:
```
if(xxx) {
 const description = `You will be charged ${currencySymbol}${formattedTotalPrice} (incl. ${taxCopy}) per ${period} for the next ${period}.<br/>After that time you will be charged a regular price of ${currencySymbol}${regularPrice}.`;
      return t(`subscription-desc-coupon-${period}`, description, {
        currencySymbol,
        formattedTotalPrice,
        taxCopy,
        period,
        regularPrice
      });
     }
```

Reply:
done as well